linked_list/linked_list.py

```diff
@@ -13,63 +13,144 @@ def __init__(self):
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: 0(1)
+    # Space Complexity: 0(n)
     def get_first(self):
-        pass
+        if self.head == None:
+            return None
+#The other scenario would mean there is a value in the first node or the 1st node exists. 
+        else:
+            return self.head.value
 
+        
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
     # Time Complexity: ?
     # Space Complexity: ?
     def add_first(self, value):
-        pass
+        if self.head == None:
+            new_node = Node(value=value, next_node=None)
+            self.head= new_node
+        else:
+            new_node = Node(value=value, next_node=self.head)
+            self.head = new_node
+
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     # Time Complexity: ?
     # Space Complexity: ?
     def search(self, value):
-        pass
+        if self.head == None:
+            return False
+        else:
+            current = self.head
+            while current != None:

```

Comment:
nice! one way to shorten this is:

```suggestion
            while current:
```